src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Configuration/IdentityConfiguration.cs

```diff
@@ -0,0 +1,391 @@
+                    SecurityTokenHandlers.Configuration.ServiceTokenResolver = new AggregateTokenResolver(new SecurityTokenResolver[] { serviceCertificateResolver, tokenResolver });
+                }
+                else
+                {
+                    SecurityTokenHandlers.Configuration.ServiceTokenResolver = serviceCertificateResolver;
+                }
+            }
+
+            SecurityTokenResolver configuredIssuerTokenResolver = IssuerTokenResolver;
+
+            if (IssuerTokenResolver == SecurityTokenHandlerConfiguration.DefaultIssuerTokenResolver)
+            {
+                //
+                // Add the known certificates from WCF's ServiceCredentials in front of 
+                // the default issuer token resolver.
+                //
+                if (KnownIssuerCertificates != null)
+                {
+                    int count = KnownIssuerCertificates.Count;
+                    if (count > 0)
+                    {
+                        SecurityToken[] tokens = new SecurityToken[count];
+                        for (int i = 0; i < count; i++)
+                        {
+                            tokens[i] = new X509SecurityToken(KnownIssuerCertificates[i]);
+                        }
+
+                        SecurityTokenResolver knownCertificateTokenResolver = SecurityTokenResolver.CreateDefaultSecurityTokenResolver(new ReadOnlyCollection<SecurityToken>(tokens), false);
+                        
+                        IssuerTokenResolver = new AggregateTokenResolver(new SecurityTokenResolver[] { knownCertificateTokenResolver, configuredIssuerTokenResolver });                       
+                    }
+                }
+            }
+            
+            if (CertificateValidationMode != X509CertificateValidationMode.Custom)
+            {
+                defaultCollection.Configuration.CertificateValidator = X509Util.CreateCertificateValidator(defaultCollection.Configuration.CertificateValidationMode,
+                                                                                                            defaultCollection.Configuration.RevocationMode,
+                                                                                                            defaultCollection.Configuration.TrustedStoreLocation);
+            }
+            else if (object.ReferenceEquals(defaultCollection.Configuration.CertificateValidator, SecurityTokenHandlerConfiguration.DefaultCertificateValidator))
+            {
+                //
+                // If the mode is custom but the validator or still default, something has gone wrong.
+                //
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ID4280)));
+            }
+
+            IsInitialized = true;
+        }
+
+        /// <summary>
+        /// TODO : Discuss better way to separate the configurations from Primitives
+        /// </summary>
```

Comment:
This should be driven by DI and the newer configuration api's. Out of scope for this change. Can you open an issue for configuring SecurityTokenHandler's via config and DI.